src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -328,22 +349,83 @@ parse_function_pointer_restrictions_from_file(
   }
   return merged_restrictions;
 }
+
+function_pointer_restrictionst::restrictionst
```

Comment:
The purpose of this needs to be documented, I am fairly sure I wrote this and it took me like 5 minutes trying to figure out what this was doing and why